packages/astro/src/core/preview/vite-plugin-astro-preview.ts

```diff
@@ -48,6 +60,13 @@ export function vitePluginAstroPreview(settings: AstroSettings): Plugin {
 					}
 				}
 
+				for (const middleware of server.middlewares.stack) {
+					// This hardcoded name will not break between Vite versions
+					if ((middleware.handle as Connect.HandleFunction).name === 'vite404Middleware') {
+						middleware.handle = handle404;
+					}
+				}
+
 				next();
```

Comment:
Was it necessary to overwrite vite's middleware _during_ the execution of our other middleware?

Reply:
@lilnasy 
Thanks, I'll do a little research to see if I can move somewhere properly handle vite404Middleware.

Reply:
It's fine if you don't find it or if it takes too long. This is already quite an improvement!

Reply:
We could add a "// TODO: look into why the replacement needs to happen here" in the meanwhile.

Reply:
Added TODO comment.
I will research and send another PR if I think it can be improved.
Thank you very much.

Reply:
If we do, we may need to set appType to 'custom' and add HTML middlewares, I think.

References:
https://github.com/vitejs/vite/blob/775bb5026ee1d7e15b75c8829e7f528c1b26c493/packages/vite/src/node/preview.ts#L213-L219
https://github.com/vitejs/vite/blob/main/packages/vite/src/node/server/middlewares/notFound.ts#L3
https://vitejs.dev/config/shared-options.html#apptype